tests/components/frontier_silicon/test_config_flow.py

```diff
@@ -294,7 +316,102 @@ async def test_invalid_device_url(
     assert result3["type"] == FlowResultType.CREATE_ENTRY
     assert result3["title"] == "Name of the device"
     assert result3["data"] == {
-        "webfsapi_url": "http://1.1.1.1:80/webfsapi",
-        "pin": "1234",
+        CONF_WEBFSAPI_URL: "http://1.1.1.1:80/webfsapi",
+        CONF_PIN: "1234",
     }
     mock_setup_entry.assert_called_once()
+
+
+async def test_ssdp(hass: HomeAssistant, mock_setup_entry: MockConfigEntry) -> None:
+    """Test a device being discovered."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_SSDP},
+        data=MOCK_DISCOVERY,
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "confirm"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {},
+    )
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "Name of the device"
+    assert result2["data"] == {
+        CONF_WEBFSAPI_URL: "http://1.1.1.1:80/webfsapi",
+        CONF_PIN: DEFAULT_PIN,
+    }
+    mock_setup_entry.assert_called_once()
+
+
+async def test_ssdp_invalid_location(hass: HomeAssistant) -> None:
+    """Test a device being discovered."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_SSDP},
+        data=INVALID_MOCK_DISCOVERY,
+    )
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
+
+
+async def test_ssdp_already_configured(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test an already known device being discovered."""
+
+    config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_SSDP},
+        data=MOCK_DISCOVERY,
+    )
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+@pytest.mark.parametrize(
+    ("webfsapi_endpoint_error", "result_error"),
+    [(ValueError, "unknown"), (ConnectionError, "cannot_connect")],
+)
+async def test_ssdp_fail(
+    hass: HomeAssistant, webfsapi_endpoint_error: Exception, result_error: str
+) -> None:
+    """Test a device being discovered but failing to reply."""
+    with patch(
+        "afsapi.AFSAPI.get_webfsapi_endpoint",
```

Comment:
For patch targets we should use the most exact target possible to ensure we are patching the one in the config flow and not the integration

Reply:
The patch target is directly called in the config flow: https://github.com/home-assistant/core/pull/89496/files#diff-e4ca90eeacfa8e4bfd0d21ab5939ca6566bfe58c7fb718af3c1e85cf98315da7R135

Reply:
I didn't want you to miss beta with this so I took care of it in 439e9b27de1858b7984d6b22920ab3ad876d2375